k8s/cmd/nsmdp/nsmdp.go

```diff
@@ -231,6 +226,78 @@ func NewNSMDeviceServer(serviceRegistry serviceregistry.ServiceRegistry) error {
 	return err
 }
 
+func (nsm *nsmClientEndpoints) addClientDevice() {
+	nsm.mutext.Lock()
+	defer nsm.mutext.Unlock()
+
+	for {
+		nsm.clientId++
+
+		id := fmt.Sprintf("nsm-%d", nsm.clientId)
+		if _, ok := nsm.devs[id]; ok {
+			// Item already exists, increment to new one.
+			continue
+		}
```

Comment:
We can close the for here.

Reply:
and can we prevent the endless loops.

Reply:
We could not close here, assume after restart we recieved nsm-1,...nsm-30, nsm-40 and all already used, so we need to add more items line nsm-41,nsm-42.